hi-ml/src/health_ml/run_ml.py

```diff
@@ -258,7 +272,7 @@ def set_trainer_for_inference(self) -> None:
         internally, which replicates some samples to make sure all devices have the same batch size in case of
         uneven inputs which biases the results."""
         mlflow_run_id = get_mlflow_run_id_from_trainer(self.trainer)
-        self.container.max_num_gpus = 1
+        self.container.max_num_gpus = self.container.max_num_gpus_inference
```

Comment:
Rather than modifying max_num_gpus in place, make it an argment to `create_lightning_trainer`?

Reply:
good point will do

Reply:
As discussed, will handle this in a separate PR.